tt_metal/impl/dispatch/command_queue.hpp

```diff
@@ -509,12 +521,20 @@ class CommandQueue {
     enum class CommandQueueMode {
         PASSTHROUGH = 0,
         ASYNC = 1,
+        TRACE = 2,
     };
     CommandQueue() = delete;
-    CommandQueue(Device* device, uint32_t id);
     ~CommandQueue();
 
+    // Command queue constructor
+    CommandQueue(Device* device, uint32_t id, CommandQueueMode mode = CommandQueue::default_mode());
+
+    // Trace queue constructor
+    CommandQueue(Trace* trace);
```

Comment:
Yes exactly, and this should not be a user visible constructor, it should be internal to the Trace object only, maybe I can use "private" and "friend class" to protect it better

Reply:
I ended up leaving it because apparently private constructors will stay private no matter what. In the next PR where I properly implement InstantiateTrace, I will likely be able to get rid of this constructor altogether because there will be no need to pass in the trace to the CQ